src/main/java/org/sagebionetworks/bridge/spring/controllers/StudyController.java

```diff
@@ -159,16 +165,37 @@ public String getAllStudies(@RequestParam(required = false) String format,
             Collections.sort(activeStudiesSummary, STUDY_COMPARATOR);
             ResourceList<Study> summaries = new ResourceList<Study>(activeStudiesSummary)
                     .withRequestParam("summary", true);
-            return Study.STUDY_LIST_WRITER.writeValueAsString(summaries);
+            return STUDY_LIST_WRITER.writeValueAsString(summaries);  
         }
         UserSession session = getAuthenticatedSession(ADMIN);
-        
         verifyCrossStudyAdmin(session.getId(), "Study admin cannot access all studies");
 
         // otherwise, return all studies including deactivated ones
         return BridgeObjectMapper.get().writeValueAsString(
                 new ResourceList<>(studies).withRequestParam("summary", false));
     }
+    
+    @GetMapping(path="/v3/studies/memberships", produces={APPLICATION_JSON_UTF8_VALUE})
+    public String getStudyMemberships() throws Exception {   
+        UserSession session = getAuthenticatedSession();
+        
+        if (session.getParticipant().getRoles().isEmpty()) {
+            throw new UnauthorizedException(STUDY_ACCESS_EXCEPTION_MSG);
+        }
+        List<String> studyIds = accountDao.getStudyIdsForUser(session.getParticipant().getSynapseUserId());
+        Stream<Study> stream = studyService.getStudies().stream();
```

Comment:
We have 30-40 studies in production. Most users will only have access to one. At most, users might have access to a handful (for example, a Sage employee might have access to mPower, CRF, PKU, and Psoriasis). It seems inefficient to do a table scan of all studies just to filter out the majority of them.

It would be better to just iterate on the studyIds and call getStudy() instead. (And special case ADMIN as needed.)

Reply:
Yes this is my bad, I thought that call was cached but only the full getStudy() call is cached. Changed this.